v-next/hardhat/src/internal/builtin-plugins/network-manager/network-manager.ts

```diff
@@ -143,8 +143,26 @@ export class NetworkManagerImplementation {
       );
 
       if (resolvedNetworkConfig.type === "edr") {
+        if (
+          resolvedChainType !== "optimism" &&
```

Comment:
Not to be resolved in this PR, but I'm starting to think that all these keywords need to be constants.